daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DaffodilParseOutputStreamContentHandler.scala

```diff
@@ -84,21 +167,39 @@ class DaffodilParseOutputStreamContentHandler(out: OutputStream, pretty: Boolean
       writer.write(System.lineSeparator())
       outputIndentation(writer)
     }
+
+    // this is for the situation where the XMLReader doesn't use the start/endPrefixMappings to
+    // pass along namespaceMapping information, so prefix information must be determined via
+    // the Attribute param.
+    // we always push but processAttributes won't always add a mapping since the Attributes param
+    // can be empty or the prefix can already exist
+    val pmAttrRes = processAttributes(atts, globalPrefixMapping)
+    globalPrefixMapping = pmAttrRes._1
+    currentElementAttributes = pmAttrRes._2
+    currentElementPrefixMapping = processAttributes(atts, currentElementPrefixMapping)._1
+
+    globalPrefixMappingContextStack.push(globalPrefixMapping)
+
     // handle start of tag
     writer.write("<")
-    writer.write(qName)
-    // handle attributes
-    for (i <- 0 until atts.getLength) {
-      val attsValue = atts.getValue(i)
-      val attsQName = atts.getQName(i)
-      writer.write(s""" $attsQName="$attsValue"""")
-    }
-    // handle namespaces
-    if (prefixMapping != null) {
-      val pm = prefixMapping.toString()
+    outputTagName(uri, localName, qName, Some(atts))
+
+    // handle namespaces in the case where prefixMappings aren't populated
+    // this contains only xmlns prefixes and are populated via the start/endPrefixMappings
+    // or Attributes via processAttributes()
+    if (currentElementPrefixMapping != null) {
```

Comment:
Not sure I understand the first line of this comment. Are there other places other than start/endPrefixMappings and Attributes where prefix mappings are populated?

Reply:
I think this was an old comment that was no longer applicable